aten/src/ATen/core/jit_type.h

```diff
@@ -1440,6 +1432,10 @@ struct CAFFE2_API ClassType : public NamedType {
     methods_.push_back(method);
   }
 
+  TypePtr withContained(std::vector<TypePtr> contained_types) {
+    return refine(std::move(contained_types));
```

Comment:
This makes the assumption that the contained_types are all subtypes of the original type.

If one of the attributes has been set to DimensionedTensorType, then is being set to TensorType (as with unshapedType) then TensorType won't subtype DimensionedTensorType

Reply:
Maybe i don't understand the use case as well